### PR TITLE
Generic pruning function (Refactoring)

### PR DESCRIPTION
**Pull request by enriavil1**
added string argument named table, which replaces kv.Receipts place.

### CODE REVIEW

File: goerli/nodekey [hunk 1]
```diff
@@ -0,0 +1 @@
+5622b144abcb8a29097a95e1c53b4eb880f2afd5e127c2921eafb13a56c30139
```
Comment (AskAlexSharov) [inline goerli/nodekey:1]: Don’t commit database to git please

Comment (AskAlexSharov) [reply goerli/nodekey:1]: https://git-scm.com/book/en/v2

Comment (enriavil1) [reply goerli/nodekey]: I am sorry. I have removed the commits with the db.

**Review by Giulio2002 — APPROVED** [review]
LGTM. @AskAlexSharov, any thoughts?

**Comment by AskAlexSharov** [comment]
I think all is good. 

Thoughts:
- Other databases have `.DeleteRange(from, to) // [from, to)` method. Such method probably can be used in pruneOldLogChunks and pruneHistoryIndex. 
- `MultiCursor` - we already have term "Dupsort Table" not sure that need introduce new naming here.
- I remember: pruneLogIndex doesn't use ETL, maybe need switch it to ETL in some future PR. 

I'm ok to merge this PR

**Comment by Giulio2002** [comment]
I just renamed MultiCursor to DupSort, i will merge after

**Comment by AlexeyAkhunov** [comment]
Sorry guys, I have introduced couple of conflicts by merging my PR ahead of this

**Comment by AlexeyAkhunov** [comment]
I have pushed the resolution into the branch `enriavil1-more-generalised-pruneReceipts`

**Pull request closed without merging**